### PR TITLE
ADDR-108: Enable i18n support for address hierarchy entries.

### PR DESCRIPTION
**Pull request by mks-d**
@mogoodrich and @mseaton 
A bit piece here making AH aware of Ext I18N for the reverse translation of address hierarchy entries.

The biggest chunk of the work was to modify `AddressHierarchyServiceImpl` while never breaking its behaviour (and unit tests) when Ext I18N is absent.

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/addresshierarchy/service/AddressHierarchyServiceImpl.java [hunk 1]
```diff
@@ -975,118 +1026,197 @@ private String encodeString(Method encodeStringMethod, String stringToEncode, St
...
+			}
+			i18nCache.setOrderedAddressFields(orderedAddressFields);
+			for (Locale locale : Context.getService(AdministrationService.class).getAllowedLocales()) {
+				initI18nCache(orderedLevels, i18nCache, locale);
+			}
+		}
+	}
+	
+	@Override
+	public void resetI18nCache() {
```
Comment (mogoodrich) [inline api/src/main/java/org/openmrs/module/addresshierarchy/service/AddressHierarchyServiceImpl.java]: Let's go ahead and move these service methods up higher in these file, before the utility methods and deprecated methods.

**Comment by mks-d** [comment]
@mogoodrich , well spotted. This is something that I changed _way_ before Ext I18n was introduced... and I never came back to it.
You're correct, the way to the full address cache is handled must be revisited to obtain the fallback you mentioned.

I guess I will refactor in such a way that the full address cache becomes accessible behind a (basic) API.

**Comment by mogoodrich** [comment]
Cool, thanks!  Let me know when you'd like me to review again!

**Comment by mks-d** [comment]
@mogoodrich , I did something pretty simple in the end. Prior to either setting to or getting from the full address cache the locale is fetched differently whether Ext I18N is loaded or not:
```java
protected Locale getAddressCacheLocaleKey() {
 if (ModuleFactory.isModuleStarted(AddressHierarchyConstants.MODULE_EXTI18N_ARTIFACT_ID)) {
  return Context.getLocale();
 }
 else {
  return LocaleUtility.getDefaultLocale();
 }
}
```
In other words, everything is stored (and retrieved) under the default locale key when Ext I18N is not around.

File: omod/src/main/resources/config.xml [hunk 2]
```diff
@@ -23,6 +23,12 @@
 
 	<aware_of_modules>
 		<aware_of_module>org.openmrs.module.legacyui</aware_of_module>
+<<<<<<< HEAD
+=======
```
Comment (mogoodrich) [inline omod/src/main/resources/config.xml]: Just noticing this... not sure what is going on here?  @mks-d

**Comment by mogoodrich** [comment]
This works for me @mks-d Can you take a look at the additional comment I just made on the config.xml, and then we should be good to go? Thanks!

**Comment by mks-d** [comment]
Thanks @mogoodrich , just changed that. Sorry, I don't know how that unmerged file sneaked in there. It didn't even break our CI because things were still fine on our master branch upon which this branch was built. Weird.

**Pull request merged**